src/main/java/io/dockstore/provision/GSPlugin.java

```diff
@@ -0,0 +1,271 @@
+                Files.createDirectories(destination.getParent());
+            }
+            catch (IOException e) {
+                System.err.println("Could not create destination path. IO exception:" + e.getMessage());
+                return false;
+            }
+
+            ProgressPrinter printer = new ProgressPrinter();
+
+            FileOutputStream fileContent;
+            try {
+                fileContent = new FileOutputStream((destination.toFile()));
+            } catch (FileNotFoundException e) {
+                System.err.println("File not found exception:" + e.getMessage());
+                return false;
+            }
+
+            PrintStream writeTo = new PrintStream(fileContent);
+
+            if (blob.getSize() < MAX_FILE_SIZE_FOR_SINGLE_WRITE) {
+                // Blob is small read all its content in one request
+                byte[] content = blob.getContent();
+                try {
+                    writeTo.write(content);
+                } catch (IOException e) {
+                    System.err.println("Could not write download file. IO exception:" + e.getMessage());
+                    return false;
+                }
+            } else {
+                // When Blob size is big or unknown use the blob's channel reader.
+                try (ReadChannel reader = blob.reader()) {
+                    WritableByteChannel channel = Channels.newChannel(writeTo);
+                    ByteBuffer bytes = ByteBuffer.allocate(DOWNLOAD_BUFFER_SIZE);
+                    try {
+                        int limit;
+                        long runningTotal = 0;
+                        while ((limit = reader.read(bytes)) > 0) {
+                            bytes.flip();
+                            channel.write(bytes);
+                            bytes.clear();
+                            runningTotal = runningTotal + limit;
+                            printer.handleProgress(runningTotal, inputSize);
+                        }
+                    } catch (IOException e) {
+                        System.err.println("Could not download file. IO exception:" + e.getMessage());
+                        return false;
+                    }
+
+                }
+            }
+
+            writeTo.close();
+            return true;
+        }
+
+        public boolean uploadTo(String destPath, Path sourceFile, Optional<String> metadata) {
+            long inputSize = sourceFile.toFile().length();
+
+            String contentType = "application/octet-stream";
+            try {
+                contentType = Files.probeContentType(sourceFile);
+            } catch (IOException e) {
+                System.err.println("Could not get content type of file. Using default content type. IO exception:" + e.getMessage());
+            }
+
+            // Bug on Mac where Files.probeContentType returns null
+            if (contentType == null)
+                contentType = "application/octet-stream";
```

Comment:
String is duplicated.

Reply:
I have to set it again because it gets set to null on a Mac when Files.probeContentType is called

Reply:
You have the string "application/octet-stream" twice. What if you make a typo in one of the instances? You should define a variable that has the string once, and use the variable twice.

Reply:
Added a static string variable to hold the content type